src/main/java/combat_system/Combat.java

```diff
@@ -0,0 +1,199 @@
+
+    Combat(ArrayList<GameCharacter> participants) {
+        this.participants = participants;
+        this.foes = participants.size()-1;
+    }
+    /**
+     * Applies a status effect if there is currently no status effect of that name applied or if there is a status
+     * effect of that name but its duration is lower than the new application's.
+     * @param target a reference to the target of the status effect
+     * @param status the status effect to apply
+     * @param duration the duration of the status effect to apply
+     */
+    public void statusEffect(GameCharacter target, StatusEffect status, int duration) {
+        if(target.getStatusEffects().get(status) == null || target.getStatusEffects().get(status)
+                < duration) {
+            target.setStatusEffect(status, duration);
+        }
+    }
+    /**
+     * Increments the round and goes through each status effect for all participants and decrements their duration
+     */
+    public void endRound() {
+        this.round += 1;
+        for (GameCharacter participant : this.participants) {
+            for(Map.Entry<StatusEffect, Integer> entry : participant.getStatusEffects().entrySet()) {
+                participant.setStatusEffect(entry.getKey(), entry.getValue()-1);
+            }
+        }
+    }
+
+    /**
+     * Used at the end of combat to clear all participants' status effects.
+     * Duration 0 causes the status to be removed, see GameCharacter.setStatusEffect for more information
+     */
+    public void clearStatus() {
+        for (GameCharacter participant : this.participants) {
+            for (Map.Entry<StatusEffect, Integer> entry : participant.getStatusEffects().entrySet()) {
+                participant.setStatusEffect(entry.getKey(), 0);
+            }
+        }
+
+    }
+
+    /**
+     * Calculates if an attack hits and the amount of damage it deals
+     * @return the string description of damage being delt
+     */
+    public String damage(int toHit, GameCharacter target, GameCharacter attacker) {
+        if(toHit >= target.getArmor().getDefense()) {
+            int health_remaining = target.getCurrentHealth()-attacker.getWeapon().getDamage();
+            target.setCurrentHealth(health_remaining);
+            if(health_remaining <= 0) {
+                if(target instanceof NonPlayerCharacter) {
+                    this.foes -= 1;
+                }
+                else {
+                    this.player_alive = false;
+                }
+            }
+
+            return "The attack hits, dealing " + attacker.getWeapon().getDamage() + " damage!";
+        }
+        else {
+            return "The attack misses!";
+        }
+    }
+
+    /**
+     * This method determines if a character is an NPC or a player and then proceeds to have them take a turn.
+     * @param participant the GameCharacter who is taking the turn
+     */
+    public void take_turn(GameCharacter participant) {
```

Comment:
maybe we should extract take_turn and combatLoop into their own class and separate player choices and NPC choice into their own methods (maybe even a class?) to make it easier to add new choices.

Reply:
Yes I agree, I will be working on that today!